include/logger.h

```diff
@@ -8,14 +8,22 @@
 
 // * Log record : each line of the log
 // * log format : tuple like <Date, Name, Message>
-// * log entity : Date, Name, Message
+// * log property : Date, Name, Message
 #define LOG(severity)                                                          \
   logger::Logger::getInstance().log<logger::severity>(                         \
-      {__func__, __FILE__, __LINE__})
+      {__FILE__, __func__, __LINE__})
+
+
+std::string hex(intptr_t v) {
+  std::stringstream ss;
```

Comment:
yea, that would be ideal, couldn't think of how to get whether the current state of ostream is std::hex, std::dec or std::oct etc. We could do:
```
std::string stream_hex(ostream ss,intptr_t v) {
   // ideally, const auto orig_state = ss.get_numeric_format_or_some_function_like_that();
  const auto orig_state << std::dec;
    return ss << std::hex << v << orig_state;
}